HDVF/include/CGAL/HDVF/Geometric_chain_complex_tools.h

```diff
@@ -0,0 +1,474 @@
+        Mesh_object_io mesh_L = Duality_simplicial_complex_tools::export_meshObject(_K) ;
+
+        // Closing K by adding the icosphere
+        //  Compute a bounding icosphere
+        Io_node_type bary = mesh_L.barycenter() ;
+        double r = mesh_L.radius(bary) ;
+        Icosphere_object_io ico(2,bary, BB_ratio*r) ;
+        ico.print_infos() ;
+
+        // Add it to the mesh
+        mesh_L.push_back(ico) ;
+        mesh_L.print_infos() ;
+
+        // Write this mesh to an off file for Tetgen
+        mesh_L.write_off(out_file_prefix) ;
+        //        const std::string tetgen_path(CMAKE_TETGEN_PATH) ;
+        // WARNING: use CGAL 3D triangulation to get rid of tetgen...
+        const std::string tetgen_path("/Users/umenohana/Dropbox/G-Mod/TopAlg/code/tetgen1.6.0/build/") ;
+        std::string tetgen_command = tetgen_path+"tetgen -pqkcYfe "+out_file_prefix+".off" ;
+        system(tetgen_command.c_str()) ;
+
+        // Read the mesh built by tetgen for L
+        Tet_object_io tetL(out_file_prefix) ;
+
+        // Build the associated SimpComplex
+        ComplexType& L = *new ComplexType(tetL, tetL.get_nodes()) ;
+
+        // Build the Sub_chain_complex_mask encoding _K inside L
+        SubCCType& K(*new SubCCType(L, false)) ;
+        // Visit all cells of _K and activate the corresponding bit in K
+        for (int q=0; q<=_K.dimension(); ++q)
+        {
+            for (size_t i=0; i<_K.number_of_cells(q); ++i)
+            {
+                const std::vector<size_t>& simplex(_K._ind2simp.at(q).at(i).get_vertices()) ;
+                const size_t id(L._simp2ind.at(q)[simplex]) ;
+                K.set_bit_on(q, id) ;
+            }
+        }
+        TripleRes t = {L,K,tetL.nodes} ;
+        return t ;
+    }
+
+    /** \brief Exports a SimpComplex to a MeshObject  */
+    static Mesh_object_io& export_meshObject(const ComplexType& _CC)
+    {
+        std::vector<Io_cell_type> vcells ;
+        for (int q = 0; q <= _CC.dimension(); ++q)
+            for (size_t i = 0; i<_CC.number_of_cells(q); ++i)
+            {
+                const Simplex& s(_CC._ind2simp.at(q).at(i)) ;
+                vcells.push_back(s.get_vertices()) ;
+            }
+
+        std::vector<Io_node_type> coords;
+        for (auto it = _CC.get_vertices_coords().begin(); it != _CC.get_vertices_coords().begin(); ++it)
+            coords.push_back(Io_node_type(*it));
+        Mesh_object_io &m = *(new Mesh_object_io(-3, coords, vcells)) ;
+        return m ;
+    }
+} ;
+
+// =========== Cubical chain complex tools
+// Build K sub-chain complex of L (homeomorphic to B^n)
+// Adjust L size
+
+/*!
+ \ingroup PkgHDVFAlgorithmClasses
+
+The class `Duality_cubical_complex_tools` is dedicated to Alexander duality for 3D binary volumes.
+
+Starting from a Cubical_chain_complex `_K`, the method `cubical_chain_complex_BB` builds a Cubical_chain_complex L and Sub_chain_complex_mask K.
```

Comment:
```suggestion
Starting from a Cubical_chain_complex `_K`, the method `cubical_chain_complex_bb()` builds a Cubical_chain_complex L and Sub_chain_complex_mask K.
```

Reply:
I think it would read nicer if you replaced Cubical_chain_complex with cubical chain complex. In case you want to keep it you must backtick, as it is a type.  Why the letter "K"?  Can you avoid the starting underscore?

Reply:
Also the different classes should better be in their own header files.